frame/ethereum/src/lib.rs

```diff
@@ -386,6 +309,88 @@ impl<T: Config> Module<T> {
 		}
 	}
 
+	fn do_transact(transaction: ethereum::Transaction) -> DispatchResultWithPostInfo {
+		ensure!(
+			fp_consensus::find_pre_log(&frame_system::Module::<T>::digest()).is_err(),
+			Error::<T>::PreLogExists,
+		);
```

Comment:
Sorry, should we move this part to `transact` dispatch call?

Since pre-log transaction processing is calling `Self::do_transact`, it looks like it will still fail at [line 171](https://github.com/paritytech/frontier/pull/324/files#diff-4a4404da981c8fc7fd4a5f9245136407d480ae2568de69e207c12066cb9473f5R171).

Reply:
Let me explain a little more, move this `ensure!` to call module firstly like:

```rs
#[weight = <T as pallet_evm::Config>::GasWeightMapping::gas_to_weight(transaction.gas_limit.unique_saturated_into())]
	fn transact(origin, transaction: ethereum::Transaction) -> DispatchResultWithPostInfo {
		ensure_none(origin)?;

               // here
               ensure!(
			fp_consensus::find_pre_log(&frame_system::Module::<T>::digest()).is_err(),
			Error::<T>::PreLogExists,
		);

		Self::do_transact(transaction)
	}
}
```
perhaps we should also change the ensure check to:

```rs
ensure!(fp_consensus::find_pre_log(&frame_system::Module::<T>::digest()).is_ok(), Error::<T>::PreLogExists);
```
This way makes sense and easy to understand.